### ISSUE TITLE
Non strict mode fails build on strict mode issues

### ISSUE THREAD

**Issue by gabrielittner** [autonomousapps/dependency-analysis-android-gradle-plugin#397]
**Build scan link**
I can send one through Slack if needed

**Plugin version**
0.73.0

**Gradle version**
7.0

**Android Gradle Plugin (AGP) version**
4.2.0

**Describe the bug**

I've disabled strict mode and changed the severity of all issues to `fail`:
```
dependencyAnalysis {
    strictMode(false)

    issues {
        all {
            onAny {
                severity('fail')
			}
		}
	}
}
```

When I run `buildHealth` I get build task failing but the output says that all modules are reported as good:
```
> Task :buildHealth FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':buildHealth'.
> Advice for root project
  Looking good! No changes needed

  Advice for project :core:json:api
  Looking good! No changes needed

  Advice for project :core:json:implementation
  Looking good! No changes needed

  Advice for project :core:network-status:api
  Looking good! No changes needed

  Advice for project :core:network-status:implementation
  Looking good! No changes needed

  Advice for project :core:network-status:testing
  Looking good! No changes needed

  Advice for project :core:network:api
  Looking good! No changes needed

  Advice for project :core:network:implementation
  Looking good! No changes needed

  Advice for project :core:play-store-review:api
  Looking good! No changes needed

  Advice for project :core:play-store-review:implementation
  Looking good! No changes needed

  Advice for project :core:play-store-review:testing
  Looking good! No changes needed
```

When I set the severity of everything except `onUsedTransitiveDependencies` I don't get the failure. Also when I enabled strict mode it fails but then reports issues on one module. Judging from that I assume that the strict mode property is considered for the printed report, but not when task success is determined.

**To Reproduce**
1. have a project which has issues in strict mode but is fine otherwise
2. set `severity("fail")` on `onUsedTransitiveDependencies`
3. run build health with strict mode off
                
**Expected behavior**

Build does not fail as because there are no issues

**Comment by autonomousapps** [comment]
Thanks for the report! I think I already see the problem. A build scan will not be necessary.

**Comment by autonomousapps** [comment]
The fix for this will be released with 0.74.0.
